### ISSUE TITLE
Memory leak in RelativeTimeFormat cache

### ISSUE THREAD

**Issue by mpeltonen** [moment/luxon#600]
Here is a shell script code to reproduce the issue with nodejs (takes about 20s to produce 'out of memory'):
```
#!/usr/bin/env bash

npm init -y && npm install luxon full-icu intl-relative-time-format

cat > leak.js <<EOF
require("intl-relative-time-format")
require("intl-relative-time-format/locale-data/fi")
const { DateTime, Settings } = require('luxon')

const dt = DateTime.local()

const toRelativeCalendarLoop = () => {
    dt.toRelativeCalendar({ base: DateTime.local(), locale: 'fi' })
    setTimeout(toRelativeCalendarLoop, 10)
}

setTimeout(toRelativeCalendarLoop, 10)
EOF

time node --max_old_space_size=10 --icu-data-dir=./node_modules/full-icu leak.js
```

The reason is that `base` DateTime is part of the `opts`, which in turn is part of the cache key, which I think it shoudn't?

Not sure if this would be the best fix for the issue, but this seems to fix it for me:
```
diff --git a/src/impl/locale.js b/src/impl/locale.js
index 2876f8c1..4b652ff7 100644
--- a/src/impl/locale.js
+++ b/src/impl/locale.js
@@ -28,7 +28,8 @@ function getCachendINF(locString, opts = {}) {

 let intlRelCache = {};
 function getCachendRTF(locString, opts = {}) {
-  const key = JSON.stringify([locString, opts]);
+  const { base, ...cacheKeyOpts } = opts
+  const key = JSON.stringify([locString, cacheKeyOpts]);
   let inf = intlRelCache[key];
   if (!inf) {
     inf = new Intl.RelativeTimeFormat(locString, opts);
```

**Comment by icambron** [comment]
Yes, that looks like the right fix to me.

**Comment by icambron** [comment]
Fixed in 836320b, will release shortly.
